controllers/service_controller.go

```diff
@@ -102,116 +103,96 @@ func (r *ServiceReconciler) translateService(ctx context.Context,
 	interfacedIdentityConfigs := make([]common.AuthConfigEvaluator, 0)
 
 	for _, identity := range service.Spec.Identity {
+		authCred := &auth_credentials.AuthCredential{
+			KeySelector: identity.Credentials.KeySelector,
+			In:          identity.Credentials.In,
+		}
 
-		var translatedIdentity config.IdentityConfig
+		translatedIdentity := &config.IdentityConfig{
+			Name:        identity.Name,
+			Credentials: authCred,
+		}
 
 		switch identity.GetType() {
-
+		// oidc
 		case configv1beta1.IdentityOidc:
-			authCred := &auth_credentials.AuthCredential{
-				KeySelector: identity.Credentials.KeySelector,
-				In:          identity.Credentials.In,
-			}
-			translatedIdentity = config.IdentityConfig{
-				OIDC: authorinoIdentity.NewOIDCIdentity(identity.Name, identity.Oidc.Endpoint, authCred),
+			translatedIdentity.OIDC = &authorinoIdentity.OIDC{
+				Endpoint:    identity.Oidc.Endpoint,
+				Credentials: authCred, // FIXME: this should not be here as the identity struct is supposed to hold the creds, not the OIDC struct
```

Comment:
Yes, it should be at Identity level... I take the blame, forgot to hoist it hehe.

Reply:
For now, let's keep copying or moving this struct into the lower level of the identity config, i.e. in the actual evaluator. It will make the code simpler, without any fancier (messier?) design using interfaces and pointers to workaround Golang's no-circular refs constraint.

We can revisit this decision in the future, say when/if other things currently stored in the generic/upper level of the identity config start being requested by the specific evaluators (e.g., the `Name` of the identity config or any other field that we end up adding). If that happens and we are forced to re-think the structures of configs and evaluators, perhaps then one option would be getting rid of the middle layer altogether (i.e. `IdentityConfig`, `MetadataConfig` and `AuthorizationConfig`) and working straight with the evaluator structs (`OIDC`, `APIKey`, `UserInfo`, `UMA`, `OPA`, etc), always copying to the lower level what is needs (thus making it self-contained, good for concurrency).

Reply:
On a second thought, this actually makes sense, it's a classic composite pattern (structural). After the call we had it was a bit clearer. We could talk about it later today.